### DISCUSSION TITLE
Mount .tsx component

### DISCUSSION THREAD

**Discussion by Dolyka** [vuejs/test-utils#2002] [Q&A] [answered]
Hi all

How can I mount a **.tsx component** using @vue/test-utils?

Used:
vitest latest
vue latest
@vue/test-utils latest

**It was found:**
If **defineComponent** is not used in the .tsx component, then the test does not fail.

File:
![image](https://user-images.githubusercontent.com/53036913/224298446-a0311a7e-4cb6-4bda-9f49-7679f87237f2.png)

Error:
![image](https://user-images.githubusercontent.com/53036913/224298251-b16681c6-30b4-4379-8b6a-99613b6e7471.png)

**Comment by cexbrayat** [comment, marked as answer]
It should work out of the box. Try to generate a project with create-vue and pick the TypeScript/JSX support to see how it is configured. If you think you found an issue, please add a reproduction so we can take a look (you can build one in a few minutes online with Stackblitz https://stackblitz.com/fork/github/vitest-dev/vitest/tree/main/examples/vue?initialPath=__vitest__ )

**Reply by Dolyka** [reply]
Thanks for the quick response.

Added reproduction on Stackblitz.
I also remind. If you remove defineComponent, then the test does not fail.

https://stackblitz.com/edit/vitest-dev-vitest-v4pacw?file=components/Spoiler.tsx

**Reply by cexbrayat** [reply]
Strange indeed, but I don't think this is an issue with VTU as we don't do anything with the components (it's Vite and the Vue plugins that handle them). 

So I would say try to open an issue with your repro on https://github.com/vitejs/vite-plugin-vue/ maybe? It looks like the jsx plugin is generating invalid code, as even a super simple component fails:

```ts
import { defineComponent } from 'vue';

export default defineComponent({
  render() {
    return 'hello';
  },
});
```

Renaming the file to `.ts` instead of `.tsx` make the test succeeds

**Comment by xenm321** [comment]
Apparently the problem is with vite. I create a project with `npm init vue@latest`, create a tsx component
![testTsx](https://user-images.githubusercontent.com/3418077/224471994-d3351f9e-e0be-4311-b1f3-c230aeb5f3e1.png)
 and test for it, with defineComponent the test fails
![vitetestfail](https://user-images.githubusercontent.com/3418077/224472142-719ab470-0c90-4a08-b2b4-688951ea57ca.png)
, without it the test passes
![withoutDefineComponent](https://user-images.githubusercontent.com/3418077/224472031-7725da81-9383-41b9-80f4-e1ad68525733.png). I created a project with vue create and create the same tsx component with defineComponent and the test passes
![tsxTestPass](https://user-images.githubusercontent.com/3418077/224472638-d994fc66-c4c1-4ebf-af96-53d038c06379.png)
